### DISCUSSION TITLE
Is it possible to assert on a list of response headers

### DISCUSSION THREAD

**Discussion by cesarsaenz** [Orange-OpenSource/hurl#2123] [Q&A] [answered]
I have a site that I want to verify by asserting header responses, however the response contains the same header name multiple times in the response with different values.  When I capture that header in a variable, its type is a list,  I then tried to use "includes and contains to look for a specific value, however those only work on strings. An example of a capture is

[Capture]
variable "headerWithMulitpleValues" header foobar

[Asserts]
variable "headerWithMulitpleValues" contains "baz"
actual: [string <name=foo; value=>, string <name=baz; value=true>]
expected : contains string <baz>
>>> types between actual and expected not consistent

Is there a way to assert a value in the header list, or a way to concert the list to a string?

Thanks

**Comment by jcamiel** [comment]
Hi @cesarsaenz 
Assuming you've an endpoint that returns multiple header "Foo", you can do the following:

```hurl
GET http://localhost:8000/assert-header
HTTP 200
[Asserts]
header "Foo" isCollection
header "Foo" count == 3
header "Foo" includes "bar"
header "Foo" includes "baz"
header "Foo" nth 0 == "bar"
header "Foo" nth 1 == "baz"
header "Foo" nth 2 == "xxx"


# Using an intermediate variable
GET http://localhost:8000/assert-header
HTTP 200
[Captures]
my_headers: header "Foo"
[Asserts]
variable "my_headers" isCollection
variable "my_headers" count == 3
variable "my_headers" includes "bar"
variable "my_headers" includes "baz"
variable "my_headers" nth 0 == "bar"
variable "my_headers" nth 1 == "baz"
variable "my_headers" nth 2 == "xxx"
```

Also, you can check what headers you're receiving with --verbose,

**Reply by cesarsaenz** [reply]
I have one more issue, the values that come back from the header are not simple types like "bar", "baz" ,but look more like bar=2; baz=3, is there a way to store those in a variable and use regex on those to further process for example
I can do 
[Asserts]
variable "my_headers" includes "bar=1; baz=3"
now I want to store this value  "bar=1; baz=3" in a variable and further process with regex to extract the value of baz which is 3 in this example or variable "my_headers" includes "bar=1; baz=3" regex on that include or can I get  the index of where the includes found that value? 
thanks again for the quick answer,

**Comment by jcamiel** [comment, marked as answer]
Maybe you can use [filters](https://hurl.dev/docs/filters.html), that allows you to transform data.
In this sample, 

```hurl
# Using an intermediate variable
GET http://localhost:8000/assert-header
HTTP 200
[Captures]
my_headers: header "Foo"
[Asserts]
variable "my_headers" nth 1 == "baz"
```

[`nth`](https://hurl.dev/docs/filters.html#nth) is a filter that extract the nth element of a list. There is a blog post introducing filters with more example here https://hurl.dev/blog/2023/01/25/hurl-2.0.0-the-graphql-edition.html#processing-data-with-filters:


```hurl
GET https://example.org/api
HTTP 200
[Asserts]
header "x-servers" split "," count == 2
header "x-servers" split "," nth 0 == "rec1.org"
header "x-servers" split "," nth 1 == "rec3.org"
jsonpath "$.books" count == 12
```

`split` allows you to transform a string to a list of substrings separated from a common string.

You have also [`regex`](https://hurl.dev/docs/filters.html#regex) and [`replace`](https://hurl.dev/docs/filters.html#replace) that you can also use to transform the value like you want.
